config/kernel-bio-op.m4

```diff
@@ -65,3 +85,28 @@ AC_DEFUN([ZFS_AC_KERNEL_BIO_BI_OPF], [
 		AC_MSG_RESULT(no)
 	])
 ])
+
+dnl #
+dnl # Linux 4.10 API,
+dnl #
+dnl # bio_set_op_attrs() was changed to an inline function instead
+dnl # of a macro.
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_BIO_SET_OP_ATTRS_IS_FUNCTION], [
+	AC_MSG_CHECKING([whether bio_set_op_attrs is a function])
+	ZFS_LINUX_TRY_COMPILE([
+		#include <linux/blk_types.h>
+	],[
+		#ifndef bio_set_op_attrs
+		(void) bio_set_op_attrs(NULL, 0, 0);
+		#else
+		#error "bio_set_op_attrs is a macro"
```

Comment:
Agreed.  The reason I did it this way was so the private definition in ```vdev_disk.c``` could stay the same (not a particularly good reason).  It seems the definition of ```bio_set_op_attrs()``` really ought to be part of ```blkdev_compat.h``` and then based on some more sane autotools checks.